src/wasm/wasm-validator.cpp

```diff
@@ -2903,7 +2903,9 @@ void FunctionValidator::visitFunction(Function* curr) {
                  "Multivalue function results (multivalue is not enabled)");
   }
   FeatureSet features;
-  // Check for things like having a rec group with GC enabled.
+  // Check for things like having a rec group with GC enabled. The type we're
+  // checking is a reference type even if this an MVP function type, so ignore
+  // the reference types feature here.
   features |=
     (Type(curr->type, Nullable).getFeatures() & ~FeatureSet::ReferenceTypes);
```

Comment:
Actually, could we do `features |= curr->type.[getParams() | getResults()]`? Then we'd include reference types when they are used in the params or results.

Reply:
Got it, thanks!